core/opendaq/component/include/opendaq/component_impl.h

```diff
@@ -700,33 +703,56 @@ ErrCode ComponentImpl<Intf, Intfs ...>::isRemoved(Bool* removed)
     return OPENDAQ_SUCCESS;
 }
 
-template <class Intf, class... Intfs>
-ErrCode INTERFACE_FUNC ComponentImpl<Intf, Intfs...>::update(ISerializedObject* obj)
+template <class Intf, class ... Intfs>
+ErrCode INTERFACE_FUNC ComponentImpl<Intf, Intfs...>::updateInternal(ISerializedObject* obj, IBaseObject* context)
 {
     const auto objPtr = SerializedObjectPtr::Borrow(obj);
+    const auto contextPtr = BaseObjectPtr::Borrow(context);
 
-    return daqTry(
-        [&objPtr, this]()
+    return daqTry([&objPtr, &contextPtr, this]
+    {
+        const bool muted = this->coreEventMuted;
+        const auto thisPtr = this->template borrowPtr<ComponentPtr>();
+        const auto propInternalPtr = this->template borrowPtr<PropertyObjectInternalPtr>();
+        if (!muted)
         {
-            const bool muted = this->coreEventMuted;
-            const auto thisPtr = this->template borrowPtr<ComponentPtr>();
-            const auto propInternalPtr = this->template borrowPtr<PropertyObjectInternalPtr>();
-            if (!muted)
-                propInternalPtr.disableCoreEventTrigger();
+            UpdatableContextPtr uptableContext = contextPtr;
+            uptableContext.setNotMutedComponent(this->globalId);
+            propInternalPtr.disableCoreEventTrigger();
+        }
 
-            const auto err = Super::update(objPtr);
+        const auto err = Super::updateInternal(objPtr, contextPtr);
 
-            updateObject(objPtr);
-            onUpdatableUpdateEnd();
+        updateObject(objPtr, contextPtr);
+        return err;
+    });
```

Comment:
This can be simplified with the change to `updateInternal`, as `disableCoreEventTrigger()` and `enableCoreEventTrigger()` should only be called on the root component of the update call. The context methods that check muted state can be removed.